### DISCUSSION TITLE
Iterative Compilation Not Working?

### DISCUSSION THREAD

**Discussion by 7Scales** [bevyengine/bevy#12676] [Q&A]
Not sure where to put this, but I haven't been able to find anything online about the issue.

I am using Windows10 and VS Code. The issue is that whenever i do "cargo run" the entirety of Bevy gets rebuilt. Also, the rust-analyzer extension seems to be having a similar problem where it checks everything instead of only new changes. This makes any sort of development virtually impossible as it can take over 10 minutes just to run the program. I've tried this on my laptop (which has the exact same setup, as far as I'm aware) with no problems at all.

I can give any other information that might help diagnose the problem.

**Comment by alice-i-cecile** [comment]
I ran into this issue a couple weeks back, driven by a bad rust-analyzer release. I would ensure that's up to date as my first step.

**Reply by msvbg** [reply]
I have also had similar problems when running a misbehaved rust-analyzer version. I'm currently running RA nightly 2024-02-17 and it seems to not cause any problems, but there's probably a stable version that works too.

**Reply by 7Scales** [reply]
I had the same version on both my desktop and laptop. After switching my desktop to the pre-release version, it seems to partially work, but it's hard to tell. It looks like RA is no longer checking absolutely everything whenever I save, but Bevy is still being completely rebuilt every time it runs, Afterwards, RA will want to check everything again.

**Comment by SkiFire13** [comment]
Are you running just `cargo run` or are you specifying some flags too (in particular `--feature`, or setting `RUSTFLAGS`)?

**Reply by 7Scales** [reply]
I did have dynamic linking enabled, along with all the other optimizations recommended by the Bevy setup guide. After encountering the issue, I got rid of all that since my laptop - which didn't have any of those things - worked fine. However, the problem still persists.
